lib/msf/base/sessions/ssh_command_shell_bind.rb

```diff
@@ -190,15 +196,16 @@ def create(params)
       mutex = Mutex.new
       condition = ConditionVariable.new
       ssh_channel = msf_channel = nil
+      opened = false
 
       if params.proto == 'tcp'
         if params.server
           raise ::Rex::BindFailed.new(params.localhost, params.localport, reason: 'TCP server sockets are not supported by SSH sessions.')
         end
 
-        ssh_channel = @ssh_connection.open_channel('direct-tcpip', :string, params.peerhost, :long, params.peerport, :string, params.localhost, :long, params.localport) do |new_channel|
+        ssh_channel = @ssh_connection.open_channel('direct-tcpip', :string, params.peerhost, :long, params.peerport, :string, params.localhost, :long, params.localport) do |_|
           dlog("new direct-tcpip channel opened to #{Rex::Socket.is_ipv6?(params.peerhost) ? '[' + params.peerhost + ']' : params.peerhost}:#{params.peerport}")
-          msf_channel = TcpClientChannel.new(self, @channel_ticker += 1, new_channel, params)
```

Comment:
These changes move the `TcpClientChannel` instance creation outside of this handler which appears to be necessary in order to read and write to the SSH channel which is required to negotiate the SSL connection.